### PR TITLE
Finish featured locations admin input

### PR DESCRIPTION
**Pull request by kshepard**

### CODE REVIEW

**Comment by flibbertigibbet** [comment]
Are we not going to do any check on the image fields to see if they actually match the dimensions?

**Comment by kshepard** [comment]
@flibbertigibbet: that's a good idea, I can add validation for that.

File: python/cac_tripplanner/destinations/models.py [hunk 1]
```diff
@@ -25,7 +26,10 @@ class Destination(models.Model):
                                help_text=('The map automatically updates as the address is typed, '
                                           'but may be overridden manually if incorrect.'))
     point = models.PointField()
-    image = models.ImageField(upload_to='destinations/', null=True)
+    image = models.ImageField(upload_to='destinations/', null=True,
```
Comment (notthatbreezy) [inline python/cac_tripplanner/destinations/models.py]: Now that we have an AWS account set up, should we create a bucket to upload images to or is that a separate task (I'm not sure what's involved with that)

Comment (kshepard) [inline python/cac_tripplanner/destinations/models.py]: That also sounds like a good idea. I'll look into that.

**Comment by flibbertigibbet** [comment]
I imagine the python tests will need updating for the changed model.

**Comment by kshepard** [comment]
Added image validation and s3 storage (configurable). This could use another review.

File: python/cac_tripplanner/destinations/migrations/0007_destination_website_url.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('destinations', '0006_add_city_state_defaults'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='destination',
+            name='website_url',
+            field=models.URLField(default='http://github.com/azavea/cac-tripplanner'),
```
Comment (flibbertigibbet) [inline python/cac_tripplanner/destinations/migrations/0007_destination_website_url.py]: Might be better to show an error to the user if they haven't entered a link, or make the field not be required; we probably don't want this link to make it into production use.

File: python/cac_tripplanner/destinations/views.py [hunk 3]
```diff
@@ -69,13 +79,9 @@ def get(self, request, *args, **kwargs):
...
-            pnt = obj['point']
-            obj['point'] = json.loads(pnt.json)
-            img = obj.get('image')
-            if img:
-                obj['image'] = img.url
-            else:
-                obj.pop('image')
+            obj['point'] = json.loads(obj['point'].json)
+            obj['image'] = self.image_to_url(obj, 'image')
+            obj['wide_image'] = self.image_to_url(obj, 'wide_image')
```
Comment (flibbertigibbet) [inline python/cac_tripplanner/destinations/views.py]: Will this work ok if an image field is null?  (Looks like they're both still nullable.)

**Comment by flibbertigibbet** [comment]
Image dimensions check looks nice.  Could we also do a file size check?  Seem to recall I managed to get an nginx error on attempting to upload a GIF.  An allowed extension message/check might be good, too.  Another thought:  we might want to run an image minimizer on the uploads.

**Comment by flibbertigibbet** [comment]
Apart from above considerations, +1

File: python/cac_tripplanner/destinations/migrations/0007_destination_website_url.py (hunk 2 above)
Comment (kshepard) [inline python/cac_tripplanner/destinations/migrations/0007_destination_website_url.py]: Ok, I can make it an optional field.

File: python/cac_tripplanner/destinations/views.py (hunk 3 above)
Comment (kshepard) [inline python/cac_tripplanner/destinations/views.py]: Yep, it works fine with nulls: image_to_url has a null check.

**Comment by kshepard** [comment]
An image size check sounds like a good idea, I'll add that. I probably don't have time to add an image minimizer at the moment, but I'll add an issue for it so we can investigate it in the future.

**Pull request merged**